OpenRA.Game/Network/Connection.cs

```diff
@@ -244,13 +244,12 @@ void IConnection.SendImmediate(IEnumerable<Order> orders)
 
 		void IConnection.SendSync(int frame, int syncHash, ulong defeatState)
 		{
-			var sync = (frame, syncHash, defeatState);
-			sentSync.Enqueue(sync);
-
 			// Send sync packets together with the next set of orders.
 			// This was originally explained as reducing network bandwidth
 			// (TCP overhead?), but the original discussions have been lost to time.
-			queuedSyncPackets.Add(OrderIO.SerializeSync(sync));
+			// Add the sync packets to the send queue before adding them to the local sync queue in the Send() method.
+			// Otherwise the client will process the local sync queue before sending the packet, and crash.
```

Comment:
This comment is no longer true!

Reply:
The question is should we keep the previous changes here at all? They are not needed with new change, but does it make the code/logic clearer?

Reply:
I think so yeah.